src/CacheManager.Core/BaseCacheManager.Keys.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using CacheManager.Core.Logging;
+
+namespace CacheManager.Core
+{
+    public sealed partial class BaseCacheManager<TCacheValue>
+    {
+        /// <inheritdoc />
+        override public IEnumerable<string> Keys(string pattern, string region)
+        {
+            CheckDisposed();
+
+            var keys = new HashSet<string>();
+
+            if (_logTrace)
+            {
+                Logger.LogTrace("Keys [{0}:{1}] started.", region, pattern);
+            }
+
+            for (var handleIndex = 0; handleIndex < _cacheHandles.Length; handleIndex++)
```

Comment:
Code changed to use the lowest level that supports keys.

see region_keys branch in my fork. (I split out my various changes into dev branches to make them more manageable) for the configuration option.

The only thing I can think of where this would be useful would be disabling the "no region" key tracking hash. Did you have something else in mind?